src/java/org/apache/cassandra/db/ReadCommand.java

```diff
@@ -578,12 +578,8 @@ private void countTombstone(ClusteringPrefix<?> clustering)
             public void onClose()
             {
                 recordReadLatency(metric, System.nanoTime() - startTimeNanos);
-
-                metric.tombstoneScannedHistogram.update(tombstones.get());
-                metric.liveScannedHistogram.update(liveRows);
-
-                if (tombstones.checkAndTriggerWarning())
```

Comment:
Can we really get rid of the guardrail check here?

Reply:
Sorry, I should have mentioned it's previously been moved into `TableMetrics.incTombstones`:

```
    public void incTombstones(long tombstones, boolean triggerWarning)
    {
        tombstoneScannedHistogram.update(tombstones);
        tombstoneScannedCounter.inc(tombstones);
        if (triggerWarning)
            tombstoneWarnings.inc();
    }
```

and `metric.liveScannedHistogram.update(liveRows)` is now in `TableMetrics.incLiveRows`